server/src/repos/RecipeRepository.ts

```diff
@@ -0,0 +1,148 @@
+import { Db, Collection, InsertOneResult, ObjectId } from "mongodb";
+import {
+  RecipePayload,
+  Recipe,
+  RecipePatchPayload,
+  RecipeCreateRequestPayload,
+} from "../interfaces/Recipe";
+
+export interface RecipeRepositoryInterface {
+  createRecipe(recipeData: RecipeCreateRequestPayload): Promise<ObjectId>;
+  getAllRecipes(): Promise<Recipe[]>;
+  getRecipeById(id: string | ObjectId): Promise<Recipe | null>;
+  updateRecipe(
+    id: string | ObjectId,
+    updatedRecipe: RecipePatchPayload
+  ): Promise<boolean>;
+  deleteRecipeById(id: string | ObjectId): Promise<boolean>;
+}
+
+export class RecipesRepository implements RecipeRepositoryInterface {
+  private collection: Collection;
+
+  constructor(private db: Db) {
+    this.collection = db.collection("recipes");
+  }
+
+  async createRecipe(
+    recipeData: RecipeCreateRequestPayload
+  ): Promise<ObjectId> {
+    try {
+      const newRecipe: RecipePayload = {
+        title: recipeData.title,
+        description: recipeData.description,
+        categories: recipeData.categories,
+        ingredients: recipeData.ingredients,
+        steps: recipeData.steps,
+        createdAt: Math.floor(Date.now() / 1000),
+      };
+      const result: InsertOneResult<any> = await this.collection.insertOne(
+        newRecipe
+      );
+      return result.insertedId as ObjectId;
+    } catch (error) {
+      console.error("Error creating recipe:", error);
+      throw error;
+    }
+  }
+
+  async getAllRecipes(): Promise<Recipe[]> {
+    try {
+      const documents = await this.collection.find().toArray();
+
+      const recipes: Recipe[] = documents.map((recipe) => ({
+        id: recipe._id,
+        title: recipe.title,
+        description: recipe.description,
+        categories: recipe.categories,
+        ingredients: recipe.ingredients,
+        steps: recipe.steps,
+        createdAt: recipe.createdAt,
+      }));
+      return recipes;
+    } catch (error) {
+      console.error("Error getting recipes:", error);
```

Comment:
I don't see any need for having a try-catch statement here. You can remove it completely because the only thing it helps you with is to just console.error() the caught error. In this case I don't think Repository should log it, since you throw it in the catch statement anyway. Moreover, your request handler logs it too already.

Reply:
Same applies for all other methods in this class.